app/utils/client/lib/canDeleteMessage.js

```diff
@@ -17,9 +17,10 @@ export const canDeleteMessage = ({ rid, ts, uid }) => {
 		return false;
 	}
 
-	const hasPermission = hasAtLeastOnePermission('delete-message', rid);
-	const deleteOwn = uid === userId;
-	if (!hasPermission && !deleteOwn) {
+	const allowed = hasPermission('delete-message', rid);
+
+	const deleteOwn = allowed || (uid === userId && hasPermission('delete-own-message'));
```

Comment:
either remove the `allowed` here or remove it from the `if` below
```suggestion
	const deleteOwn = uid === userId && hasPermission('delete-own-message');
```

Reply:
I'm skipping one hasPermission, if you are allowed any messages you are already allowed to edit your own messages... we save 3~4 finds on the database...